internal/core/domain/swap.go

```diff
@@ -0,0 +1,37 @@
+package domain
+
+import (
+	"context"
+	"time"
+
+	"github.com/ArkLabsHQ/fulmine/pkg/boltz"
+)
+
+type SwapStatus int
+
+const (
+	SwapPending SwapStatus = iota
+	SwapFailed
+	SwapSuccess
+)
+
+type Swap struct {
+	Id         string
+	Amount     uint64
+	Date       time.Time
+	To         boltz.Currency
+	From       boltz.Currency
+	Status     SwapStatus
+	Invoice    string
+	VHltcId    string // If the swap is outbound, this is the Sent VHTLC Id
+	RedeemTxId string // If the swap is inbound, this is the Redeem Tx Id, else this is ID of reclaim Failed Outbound Swap
+}
```

Comment:
nice abstraction, we can make it include the vhtlc.Opts and get rid of the vhtlc opts repository interface (let's keep the sqlite migrations and queries):
```suggestion
type Swap struct {
	Id         string
	Amount     uint64
	Date       time.Time
	To         boltz.Currency
	From       boltz.Currency
	Status     SwapStatus
	Invoice    string
	Params    vhtlc.Opts
	FundTxid string // the txid of the virtual tx that funded the vhtlc
	RedeemTxId string // the txif of the virtual tx that redeemed the funds, by either "claiming" or "refunding"
}
```

Reply:
in the sqlite impl of the below interface, we should take care of storing the opts in a dedicated table: basically we need to move that code in the swap repo and make the inserts transactional.

This is an [example](https://github.com/ark-network/ark/blob/master/server/internal/infrastructure/db/sqlite/round_repo.go#L67-L182) taken from arkd that can be used as reference: 2 tables are updated when adding or updating a round - `round` and `tx`; note how both writes are made transactional by the mean of `execTx`.

Reply:
The function `GetVHTLC` which pertains to the construction of VHTLC and also the storage of vhtlc options is not only used in `IncreaseOutboutCapacity` and `IncreaseInboundCapacity` functions, but there exist also a call from the GRPC handler function `CreateVHTLC` which also makes use of said function. 

If a transaction is supposed in which VHTLC opts  updates are tied to swaps update, which I assume is the goal you had in mind. That unfortunately,  would have a repercussion to the `CreateVHTLC function`.